llm_perf/common/benchmark_runner.py

```diff
@@ -0,0 +1,115 @@
+import os
+import traceback
+from abc import ABC, abstractmethod
+from itertools import product
+from logging import getLogger
+from typing import Any, Dict, List, Optional
+
+from llm_perf.common.utils import (
+    CANONICAL_PRETRAINED_OPEN_LLM_LIST,
+    OPEN_LLM_LIST,
+    PRETRAINED_OPEN_LLM_LIST,
+)
+from optimum_benchmark import Benchmark, BenchmarkConfig, BenchmarkReport
+from optimum_benchmark.logging_utils import setup_logging
+
+
+class BenchmarkRunner(ABC):
```

Comment:
not a fan of using the keyword runner here, maybe an LLMPerfBenchmarkManager, no strong opinion on this tho